tests/specs/mcd/flipper-bids-pass-rough-spec.k

```diff
@@ -0,0 +1,233 @@
+     andBool #rangeUInt(256, VChainId)
+
+     andBool (#rangeUInt(256, ABI_n)
+     andBool (#rangeUInt(256, Bid)
+     andBool (#rangeUInt(256, Lot)
+     andBool (#rangeAddress(Guy)
+     andBool (#rangeUInt(48, Tic)
+     andBool (#rangeUInt(48, End)
+     andBool (#rangeAddress(Usr)
+     andBool (#rangeAddress(Gal)
+     andBool (#rangeUInt(256, Tab)
+     andBool (#sizeByteArray(CD) <=Int 1250000000
+     andBool (#notPrecompileAddress(Guy)
+     andBool (#notPrecompileAddress(Usr)
+     andBool (#notPrecompileAddress(Gal)
+     andBool (VGas >=Int 3000000
+     andBool (#rangeUInt(256, Junk_0)
+     andBool (#rangeUInt(256, Junk_1)
+     andBool (#rangeUInt(256, Junk_2)
+     andBool (#rangeUInt(256, Junk_3)
+     andBool (#rangeUInt(256, Junk_4)
+     andBool (#rangeUInt(256, Junk_5)
+     andBool ((VCallValue ==Int 0))))))))))))))))))))))
+
+     andBool #lookup(ACCT_ID_STORAGE, #Flipper.bids[ABI_n].bid) ==Int Bid
+     andBool #lookup(ACCT_ID_STORAGE, #Flipper.bids[ABI_n].lot) ==Int Lot
+     andBool #lookup(ACCT_ID_STORAGE, #Flipper.bids[ABI_n].guy_tic_end) ==Int #WordPackAddrUInt48UInt48(Guy, Tic, End)
+     andBool #lookup(ACCT_ID_STORAGE, #Flipper.bids[ABI_n].usr) ==Int Usr
+     andBool #lookup(ACCT_ID_STORAGE, #Flipper.bids[ABI_n].gal) ==Int Gal
+     andBool #lookup(ACCT_ID_STORAGE, #Flipper.bids[ABI_n].tab) ==Int Tab
+     andBool #lookup(ACCT_ID_ORIG_STORAGE, #Flipper.bids[ABI_n].bid) ==Int Junk_0
+     andBool #lookup(ACCT_ID_ORIG_STORAGE, #Flipper.bids[ABI_n].lot) ==Int Junk_1
+     andBool #lookup(ACCT_ID_ORIG_STORAGE, #Flipper.bids[ABI_n].guy_tic_end) ==Int Junk_2
+     andBool #lookup(ACCT_ID_ORIG_STORAGE, #Flipper.bids[ABI_n].usr) ==Int Junk_3
+     andBool #lookup(ACCT_ID_ORIG_STORAGE, #Flipper.bids[ABI_n].gal) ==Int Junk_4
+     andBool #lookup(ACCT_ID_ORIG_STORAGE, #Flipper.bids[ABI_n].tab) ==Int Junk_5
+     andBool #Flipper.bids[ABI_n].bid =/=Int #Flipper.bids[ABI_n].lot
+     andBool #Flipper.bids[ABI_n].bid =/=Int #Flipper.bids[ABI_n].guy_tic_end
+     andBool #Flipper.bids[ABI_n].bid =/=Int #Flipper.bids[ABI_n].usr
+     andBool #Flipper.bids[ABI_n].bid =/=Int #Flipper.bids[ABI_n].gal
+     andBool #Flipper.bids[ABI_n].bid =/=Int #Flipper.bids[ABI_n].tab
+     andBool #Flipper.bids[ABI_n].lot =/=Int #Flipper.bids[ABI_n].guy_tic_end
+     andBool #Flipper.bids[ABI_n].lot =/=Int #Flipper.bids[ABI_n].usr
+     andBool #Flipper.bids[ABI_n].lot =/=Int #Flipper.bids[ABI_n].gal
+     andBool #Flipper.bids[ABI_n].lot =/=Int #Flipper.bids[ABI_n].tab
+     andBool #Flipper.bids[ABI_n].guy_tic_end =/=Int #Flipper.bids[ABI_n].usr
+     andBool #Flipper.bids[ABI_n].guy_tic_end =/=Int #Flipper.bids[ABI_n].gal
+     andBool #Flipper.bids[ABI_n].guy_tic_end =/=Int #Flipper.bids[ABI_n].tab
+     andBool #Flipper.bids[ABI_n].usr =/=Int #Flipper.bids[ABI_n].gal
+     andBool #Flipper.bids[ABI_n].usr =/=Int #Flipper.bids[ABI_n].tab
+     andBool #Flipper.bids[ABI_n].gal =/=Int #Flipper.bids[ABI_n].tab
```

Comment:
A bit surprised this storage address pairwise disequality check is necessary...shouldn't these all be of the form `hash + offset` where `hash` stays the same but `offset` is different for every slot? Not saying there's anything wrong with it, just surprised `K` doesn't recognize that they aren't the same.

Reply:
I started by adding lemmas about the disequalities of hashes on different inputs and at different offsets, but a discussion in #902 with @daejunpark made me realize it may be easy to add lemmas which _seem_ OK, but are not.

So I switched to generating exactly the disequalities needed in the proofs in #904 (and made KLab generate these disequalities directly on my branch `pr` of my fork). Because this (i) didn't seem to impact performance at all, (ii) was easy to make KLab do correctly, and (iii) expresses _exactly_ the disequalities needed for a given proof rather than the over-approximation given by the hash2 lemmas, it seemed like the better way to go.

Reply:
Specifically if you want to see the type of `hash2` disequalities that were removed, check out: https://github.com/kframework/evm-semantics/pull/902/commits/7d81a7bcaca9eb6c107566ebab6b1c99feada14b